test/index.js

```diff
@@ -970,6 +972,36 @@ describe('DDWAF', () => {
     waf.dispose()
   })
 
+  it('should perform correct marshalling of ddwaf_object', () => {
+    const waf = new DDWAF(rules, 'recommended')
+    const context = waf.createContext()
+    const result = context.run({
+      persistent: {
+        'server.request.headers.no_cookies': 'marshalling'
+      }
+    }, TIMEOUT)
+
+    assert.strictEqual(result.timeout, false)
+    assert.strictEqual(result.status, 'match')
+    assert.strictEqual(typeof result.keep, 'boolean')
+    assert.strictEqual(result.keep, true)
+    assert.strictEqual(typeof result.attributes['_dd.appsec.trace.integer'], 'number')
+    assert.strictEqual(result.attributes['_dd.appsec.trace.integer'], 662607015)
+    assert.strictEqual(typeof result.attributes['_dd.appsec.trace.negative_integer'], 'number')
+    assert.strictEqual(result.attributes['_dd.appsec.trace.negative_integer'], -662607015)
+    assert.strictEqual(typeof result.attributes['_dd.appsec.trace.float'], 'number')
+    assert.strictEqual(result.attributes['_dd.appsec.trace.float'], 2.71828)
+    assert.strictEqual(typeof result.attributes['_dd.appsec.trace.negative_float'], 'number')
+    assert.strictEqual(result.attributes['_dd.appsec.trace.negative_float'], -3.14159)
+    assert.strictEqual(typeof result.attributes['_dd.appsec.trace.bool'], 'boolean')
+    assert.strictEqual(result.attributes['_dd.appsec.trace.bool'], true)
+    assert.strictEqual(typeof result.attributes['_dd.appsec.trace.string'], 'string')
+    assert.strictEqual(result.attributes['_dd.appsec.trace.string'], 'Lorem ipsum sit amet')
```

Comment:
not sure the typeofs are super useful here ? if they're strictly equal to these values, the typeof should be implicitely checked anyway
```suggestion
    assert.strictEqual(result.keep, true)
    assert.strictEqual(result.attributes['_dd.appsec.trace.integer'], 662607015)
    assert.strictEqual(result.attributes['_dd.appsec.trace.negative_integer'], -662607015)
    assert.strictEqual(result.attributes['_dd.appsec.trace.float'], 2.71828)
    assert.strictEqual(result.attributes['_dd.appsec.trace.negative_float'], -3.14159)
    assert.strictEqual(result.attributes['_dd.appsec.trace.bool'], true)
    assert.strictEqual(result.attributes['_dd.appsec.trace.string'], 'Lorem ipsum sit amet')
```